ckanext/pigma_theme/scripts/import.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# coding: utf8
+import argparse
+import csv
+import urllib2
+import urllib
+import json
+import ssl
+from base64 import encodestring
+
+from pygments import highlight
+from pygments.lexers import PythonLexer
+from pygments.formatters import Terminal256Formatter
+from pprint import pformat
+
+
+def pprint_color(obj):
+    print highlight(pformat(obj), PythonLexer(), Terminal256Formatter())
+
+
+def _groups_from_csv(path):
+    """
+    Read a CSV file defining the groups to create
+    :param path: path to the csv file (local, does not handle URLs)
+    :return: a list of dicts, one dict per group
+    """
+    dataset_dicts = []
+    with open(path, 'rb') as csvfile:
+        reader = csv.reader(csvfile, delimiter=',', quotechar='"')
+        # This skips the first row of the CSV file.
+        next(reader)
+        for row in reader:
+            row_as_dict = {
+                "name": row[0],
+                "title": row[1],
+                "id": row[0],
+                "image_url": row[2]
+            }
+            dataset_dicts.append(row_as_dict)
+    return dataset_dicts
+
+
+def _create_groups(dataset_dicts, API_URI, BASE64AUTH):
+    """
+    Given en list of dicts defining the groups, creates the group using CKAN's API
+    Since geOrchestra, for now, doesn't let in using the API key, we use basic auth.
+    If a group already exists in CKAN, pruges it before creating it again.
+    :param dataset_dicts:
+    :param API_URI: URL to ckan api (e.g.https://georchestra.mydomain.org/ckan/api/3/)
+    :param BASE64AUTH: basic auth chain built with user/passwd
+    :return:
+    """
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+
+    for dataset_dict in dataset_dicts:
+        try:
+            delete_req = urllib2.Request(API_URI + 'action/group_purge')
+            #delete_req.add_header('Authorization', AUTH_TOKEN)
```

Comment:
Can you delete the line ?

Reply:
I won't. See it as a reminder that when we get the API working as it should, we will make the call using API keys instead of basic auth.